/**
 * Copyright (c) 2022 Huawei Device Co., Ltd.
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

syntax = "proto3";
package protoPanda;

import "assemblyFunction.proto";
import "assemblyRecord.proto";
import "assemblyType.proto";
import "assemblyLiterals.proto";

message Program {
    message RecordTable  {
        bytes key = 1;
        Record value = 2;
    }
    message FunctionTable  {
        bytes key = 1;
        Function value = 2;
    }
    message FunctionSynnoyms  {
        bytes key = 1;
        repeated bytes value = 2;
    }
    message LiteralArrayTable {
        bytes key = 1;
        LiteralArray value = 2;
    }
    uint32 lang = 1;
    repeated RecordTable recordTable = 2;
    repeated FunctionTable functionTable = 3;
    repeated FunctionSynnoyms functionSynonyms = 4;
    repeated LiteralArrayTable literalArrayTable = 5;
    repeated bytes strings = 6;
    repeated Type arrayTypes = 7;
}
